test/overlay.html

```diff
@@ -341,6 +341,168 @@
+                it('should be aligned relative to top-right corner', done => {
+                  listenOnce(overlay, 'vaadin-overlay-open', () => {
+                    expect(overlay.hasAttribute('right-aligned')).to.be.false;
+                    expect(overlay.hasAttribute('bottom-aligned')).to.be.false;
+                    expect(overlay.style.right).to.be.equal(menu._phone ? '' : `${viewWidth - 450}px`);
+                    expect(overlay.style.top).to.be.equal(menu._phone ? '' : '10px');
+                    done();
+                  });
+                  contextmenu(450, 10);
+                });
+
+                it('should be aligned relative to bottom-right corner', done => {
+                  listenOnce(overlay, 'vaadin-overlay-open', () => {
+                    expect(menu._phone || overlay.hasAttribute('right-aligned')).to.be.false;
+                    expect(menu._phone || overlay.hasAttribute('bottom-aligned')).to.be.true;
+                    expect(overlay.style.right).to.be.equal(menu._phone ? '' : '0px');
+                    expect(overlay.style.bottom).to.be.equal(menu._phone ? '' : '0px');
+                    done();
+                  });
+                  contextmenu(viewWidth, viewHeight);
+                });
+              }
+
+              it('css should be correctly configured to set content position', done => {
+                listenOnce(overlay, 'vaadin-overlay-open', () => {
+                  const rect = content.getBoundingClientRect();
+                  expect(rect.width).to.be.closeTo(100, 0.5);
+                  expect(rect.height).to.be.closeTo(100, 0.5);
+                  !menu._phone && expect(rect.left).to.be.closeTo(viewWidth - 100, 0.5);
+                  !menu._phone && expect(rect.top).to.be.closeTo(viewHeight - 100, 0.5);
+                  done();
+                });
+                contextmenu(viewWidth, viewHeight);
+              });
+
+              it('should reset css properties and attributes on each open', done => {
+                listenOnce(overlay, 'vaadin-overlay-open', () => {
+
+                  overlay.opened = false;
+                  listenOnce(overlay, 'vaadin-overlay-open', () => {
+                    const rect = content.getBoundingClientRect();
+                    !menu._phone && expect(rect.left).to.be.closeTo(16, 0.5);
+                    !menu._phone && expect(rect.top).to.be.closeTo(16, 0.5);
+                    expect(overlay.hasAttribute('right-aligned')).to.be.true;
+                    expect(overlay.hasAttribute('bottom-aligned')).to.be.false;
+                    expect(overlay.style.right).to.be.empty;
+                    expect(overlay.style.bottom).to.be.empty;
+                    done();
+                  });
+                  contextmenu(16, 16);
+
+                });
+                contextmenu(viewWidth, viewHeight);
+              });
+
+              it('overlay position should be constrained to the viewport', done => {
```

Comment:
I guess there is no need to duplicate the test. (Related to the tests above as well) We can use extra parameter to check both directions, smth like https://github.com/vaadin/vaadin-tabs/blob/656ca19fccf034dd89240f35757cd21b9059693a/test/nav.html#L94

Reply:
Done.